src/test/camelkjsonutils.test.ts

```diff
@@ -48,7 +48,11 @@ suite('ensure camelk utilities work as expected', function() {
 					assert.ok(output.length > 0);
 					done();
 				});
-		});
+		})
+		.catch((err) => {
```

Comment:
why do we need to put the catch here although there are not in the following test "should be able to construct usable rest URL" ?

(I really don't get how Typescript is handling their exception mechanism)

Reply:
The catch is for the stringify call because it can be rejected. In the followed test I don't see a stringify? Or do I miss something?

Reply:
no, there is no stringify.
You are nto missing something, I just don't understand the philosophy of this language :-)

Reply:
As far as I understood it a Promise is handled that if you resolve it, it will return the the .then() section of the calling function. If the promise gets rejected it will end up in the .catch() block of the calling function. I am not saying I am fully understanding that language and I might be wrong too. :D  Have a look at https://basarat.gitbooks.io/typescript/docs/promise.html